pkg/requests/requests.go

```diff
@@ -0,0 +1,155 @@
+package requests
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+)
+
+const (
+	apiRepositories = "_apis/git/repositories"
+	apiProjects     = "_apis/projects"
+	itemsPath       = "items"
+	refsPath        = "refs"
+)
+
+// GetAndParseJSON invokes a HTTP request with basic auth.
+// On success the response body will be unmarshalled as JSON.
+func GetAndParseJSON(result interface{}, user, token string, urlPath *url.URL) error {
+	body, err := getBodyFromRequest(user, token, urlPath)
+	if err != nil {
+		return err
+	}
+	err = json.Unmarshal(body, &result)
+
+	return err
+}
+
+// GetAsString invokes a HTTP request with basic auth.
+// Returns the response as a string.
+func GetAsString(user, token string, urlPath *url.URL) (string, error) {
+	body, err := getBodyFromRequest(user, token, urlPath)
+	if err != nil {
+		return "", err
+	}
+
+	return string(body), nil
+}
+
+func getBodyFromRequest(user string, token string, urlPath *url.URL) ([]byte, error) {
+	url := urlPath.String()
+	req, err := http.NewRequest(http.MethodGet, url, nil)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	req.SetBasicAuth(user, token)
+
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return []byte{}, err
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return []byte{}, fmt.Errorf("unable to get. status code: %d", resp.StatusCode)
+	}
```

Comment:
Suggest checking for all 2xx codes instead of only 200.

Also, for the error it might be nice to use the `resp.Status` as that also includes the status code name that was provided by the server. Such as `"200 OK"` or `"400 Bad Request"`

```suggestion
	if resp.StatusCode < 200 || resp.StatusCode >= 300 {
		return []byte{}, fmt.Errorf("unable to get. status code: %s", resp.Status)
	}
```